includes/Story_Renderer/HTML.php

```diff
@@ -215,6 +216,23 @@ protected function replace_html_head( $content ) {
 		return $content;
 	}
 
+	/**
+	 * Force home urls to http / https based on context.
+	 *
+	 * @param string $content String to replace.
+	 *
+	 * @return string
+	 */
+	protected function replace_url_scheme( $content ) {
+		$current = is_ssl() ? 'https' : 'http';
+		$new     = is_ssl() ? 'http' : 'https';
+		$home    = get_home_url();
+
+		$content = str_replace( set_url_scheme( $home, $new ), set_url_scheme( $home, $current ), $content );
```

Comment:
I had a hard time fully grasping what these variables mean to achieve due to unclear wording. Something like `$current` and `$actual` might be better suited. Can we make this a bit easier to read?

Maybe something like this:

```suggestion
		if ( is_ssl() ) {
			// If serving via HTTPS, all content should be referenced via HTTPS.
			return str_replace( set_url_scheme( $home, 'http' ), set_url_scheme( $home, 'https' ), $content );
		}

		// TODO: Is this really needed? Serving assets over HTTPS when accessing story over HTTP is not problematic.
		// We could just do `return $content;`
		return str_replace( set_url_scheme( $home, 'https' ), set_url_scheme( $home, 'http' ), $content );

```

Reply:
I don't think this is a good change. I think we should improve the comment. Making the code longer and more to excute is not the right course. If it takes a minute to understand, that is a failure in comments and variable names, not the code.

Reply:
Even with the latest changes it's not super obvious IMO. Don't get me wrong, the code is super clean and DRY, but IMO it's not as easy to comprehend as something simple as 

```php
if ( is_ssl() ) {
	// If serving via HTTPS, all content should be referenced via HTTPS.
	return str_replace( set_url_scheme( home_url(), 'http' ), set_url_scheme( home_url(), 'https' ), $content );
}
```

just my two cents :)

Reply:
I tweaked it, just to run on SSL.